src/Microsoft.ML.TimeSeries/SrCnnEntireAnomalyDetector.cs

```diff
@@ -309,6 +309,15 @@ public void Process()
                     _previousBatch = _previousBatch.GetRange(_batch.Count, _bLen);
                     _previousBatch.AddRange(_batch);
                     _modeler.Train(_previousBatch.ToArray(), ref _results);
+
+                    // move the values to front
+                    for (int i = 0; i < _batch.Count; ++i)
```

Comment:
This is only for the first batch right

Reply:
This code snippet is for another bug, which is the result not aligned for the last batch. It is not related with the reported case.

---
In reply to: [527820005](https://github.com/dotnet/machinelearning/pull/5502#discussion_r527820005) [](ancestors = 527820005)